### PR TITLE
Precision support

### PR DESCRIPTION
**Pull request by neon60**

### CODE REVIEW

**Comment by saadrahim** [comment]
Please make CI pass.

File: docs/about/compatibility/precission-support.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+# Native hardware support
+
+## Hardware accelerated compute units support
+
+| Data types   | int8  | int16  | int32  | int64  | float8 (E4M3) | float8 (E5M2) | float16 | bfloat16 | tensorfloat32 | float32 | float64 |
+|:-------------|:-----:|:------:|:------:|:------:|:-------------:|:-------------:|:-------:|:--------:|:-------------:|:-------:|:-------:|
+| Mi100        | ✅    |  ✅    | ✅    | ✅    | ❌           | ❌           | ✅      | ✅       | ❌           | ✅      | ✅     |
+| Mi200 series | ✅    |  ✅    | ✅    | ✅    | ❌           | ❌           | ✅      | ✅       | ❌           | ✅      | ✅     |
+| Mi300 series | ✅    |  ✅    | ✅    | ✅    | ❌           | ❌           | ✅      | ✅       | ❌           | ✅      | ✅     |
```
Comment (MKKnorr) [inline docs/about/compatibility/precission-support.md:55]: Please check these again. Afaik `int64` is emulated and therefore runs at a lower rate than `int32`. Also the MI100 has half the float64 throughput as float32, but I wouldn't say, that it doesn't have "hardware acceleration"

File: docs/about/compatibility/precission-support.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+| rocRAND ({doc}`details<rocrand:data-type-support>`)                   | -/✅  | -/✅   | -/✅  | -/✅   |
+| hipRAND ({doc}`details<hiprand:data-type-support>`)                   | -/⚠️  | -/✅   | -/✅  | -/✅   |
+| rocPRIM ({doc}`details<rocprim:data-type-support>`)                   | ✅/✅ | ✅/✅ | ✅/✅ | ✅/✅ |
+| hipCUB ({doc}`details<hipcub:data-type-support>`)                     | ✅/✅ | ✅/✅ | ✅/✅ | ✅/✅ |
+| rocThrust ({doc}`details<rocthrust:data-type-support>`)               | ✅/✅ | ✅/✅ | ✅/✅ | ✅/✅ |
+| MIGraphX                                                              | ✅/❌ | ❌/❌ | ❌/❌ | ❌/❌ |
+
+### Floating points
+
+| <div style="width:180px">Library input/output floating point data types</div> | float8 (E4M3) | float8 (E5M2) | float16 | bfloat16 | tensorfloat32 | float32 | float64 |
```
Comment (MKKnorr) [inline docs/about/compatibility/precission-support.md:102]: "Library input/output floating point data types"

This is being rendered in a darker colour in the generated html file

File: docs/about/compatibility/precission-support.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+| rocRAND ({doc}`details<rocrand:data-type-support>`)                           | -/❌         | -/❌          |  -/✅  | -/❌     | -/❌         | -/✅    | -/✅    |
+| hipRAND ({doc}`details<hiprand:data-type-support>`)                           | -/❌         | -/❌          |  -/⚠️  | -/❌     | -/❌         | -/✅    | -/✅    |
+| rocPRIM ({doc}`details<rocprim:data-type-support>`)                           | ❌/❌        | ❌/❌        | ✅/✅  | ✅/✅   | ❌/❌        | ✅/✅  | ✅/✅  |
+| hipCUB ({doc}`details<hipcub:data-type-support>`)                             | ❌/❌        | ❌/❌        | ✅/✅  | ✅/✅   | ❌/❌        | ✅/✅  | ✅/✅  |
+| rocThrust ({doc}`details<rocthrust:data-type-support>`)                       | ❌/❌        | ❌/❌        | ⚠️/⚠️  | ❌/❌   | ❌/❌        | ✅/✅  | ✅/✅  |
+| MIGraphX                                                                      | ❌/❌        | ❌/❌        | ❌/❌  | ✅/✅   | ❌/❌        | ✅/✅  | ✅/✅  |
+
+## Libraries internal calculations type support
+
+The ROCm™ components support of specific internal data type is listed in the following two tables. For detailed description open the library data type support page.
```
Comment (MKKnorr) [inline docs/about/compatibility/precission-support.md:114]:
```suggestion
The ROCm™ components support of specific internal data types is listed in the following two tables. For detailed description open the library data type support page.
```

**Comment by MKKnorr** [comment]
Also please fix the spellchecker/add to the wordlist :)

File: docs/about/compatibility/precission-support.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  <meta name="keywords" content="int8, float8, float8 (E4M3), float8 (E5M2), float16, half, bfloat16, tensorfloat32, float, float32, float64, double">
+</head>
+
+# ROCm data type specifications
+
+## Integer
+
+| Data types name | HIP type                  | Description                           |
+|:----------------|:-------------------------:|:-------------------------------------:|
+| int8            | ``int8_t``,``uint8_t``    | An unsigned or signed 8-bit integer.  |
```
[the remark this answers is not in the document]
Comment (neon60) [reply docs/about/compatibility/precission-support.md:13]: I think, if a ROCm library wants to distinguish the two types, we can highlight this on the project specific page.

File: docs/about/compatibility/precission-support.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+| int32           | ``int32_t``,``uint32_t``  | An unsigned or signed 32-bit integer. |
+| int64           | ``int64_t``,``uint64_t``  | An unsigned or signed 64-bit integer. |
+
+## Floating point
+
+| Data types name | HIP type      | Description  |
+|:----------------|:-------------:|:------------:|
+| float8 (E4M3)   | -             | A 8-bit floating point number mostly following IEEE-754 conventions and bit layout **S1E4M3** described in [8-bit Numerical Formats for Deep Neural Networks](https://arxiv.org/abs/2206.02915), with expanded range and with no infinity or signed zero. NaN is represented as negative zero. |
+| float8 (E5M2)   | -             | A 8-bit floating point number mostly following IEEE-754 conventions and bit layout **S1E5M2** described in [8-bit Numerical Formats for Deep Neural Networks](https://arxiv.org/abs/2206.02915), with expanded range and with no infinity or signed zero. NaN is represented as negative zero. |
+| float16         | ``half``     | A 16-bit floating-point number, which is conform to the IEEE 754-2008 single precision storage format. |
```
[the remark this answers is not in the document]
Comment (neon60) [reply docs/about/compatibility/precission-support.md:24]: Yes. It was added at IEEE 754-2008.

File: docs/about/compatibility/precission-support.rst [hunk 6]
```diff
@@ -0,0 +1,544 @@
...
+        - ✅
+        - ✅
+      * 
+        - Mi300 series
+        - ✅
+        - ✅
+        - ✅
+        - ✅
+
+  .. tab-item:: Floating-points types
```
Comment (MKKnorr) [inline docs/about/compatibility/precission-support.rst:158]:
```suggestion
  .. tab-item:: Floating-point types
```

Also occures at several other places

File: docs/about/compatibility/precission-support.rst [hunk 7]
```diff
@@ -0,0 +1,544 @@
...
+    :sync: floating-point-type
+
+    .. list-table::
+      :header-rows: 1
+
+      * 
+        - Floating-point types name
+        - float8 (E4M3)
+        - float8 (E5M2)
```
Comment (MKKnorr) [inline docs/about/compatibility/precission-support.rst:167]: Not really important, but something that annoys me: Can the names of the float8 types be force to be rendered on one line? Then the site wouldn't jump around, when switching from integral types to floating-point types.

File: docs/about/compatibility/precission-support.rst [hunk 8]
```diff
@@ -0,0 +1,544 @@
...
+==========================================
+
+ROCm™ components support for int8, float8 (E4M3), float8 (E5M2), int16, float16,
+bfloat16, int32, tensorfloat32, float32, int64 and float64 is listed in the
+following tables.
+
+Libraries input/output type support
+-------------------------------------------------------------------------------
+
+The ROCm™ components support of specific input and output data type is listed in
```
Comment (MKKnorr) [inline docs/about/compatibility/precission-support.rst:369]:
```suggestion
The ROCm™ components support of specific input and output data types is listed in
```

File: docs/about/compatibility/precission-support.rst [hunk 9]
```diff
@@ -0,0 +1,544 @@
...
+the following two tables. For detailed description open the library data type
```
Comment (MKKnorr) [inline docs/about/compatibility/precission-support.rst:370]:
```suggestion
the following two tables. For a detailed description open the library data type
```

Also occurs in the following section

File: docs/sphinx/_toc.yml.in [hunk 10]
```diff
@@ -96,6 +96,8 @@ subtrees:
     title: LLVM ASan
   - file: conceptual/cmake-packages.rst
     title: Using CMake
+  - file: about/compatibility/precission-support.md
```
Comment (MKKnorr) [inline docs/sphinx/_toc.yml.in:99]:
```suggestion
  - file: about/compatibility/precission-support.rst
```

File: docs/about/compatibility/precission-support.rst [hunk 11]
```diff
@@ -0,0 +1,544 @@
...
+.. tab-set::
+
+  .. tab-item:: Integer types
+    :sync: integer-type
+
+    .. list-table::
+      :header-rows: 1
+
+      * 
+        - Integer types name
```
[the remark this answers is not in the document]
Comment (neon60) [reply docs/about/compatibility/precission-support.rst]: Fixed.

File: docs/about/compatibility/precission-support.rst [hunk 12]
```diff
@@ -0,0 +1,544 @@
...
+
+Hardware accelerated compute units support
+-------------------------------------------------------------------------------
+
+Integer types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tab-set::
+
+  .. tab-item:: Integer types
```
[the remark this answers is not in the document]
Comment (neon60) [reply docs/about/compatibility/precission-support.rst]: Fixed.

File: docs/about/compatibility/precission-support.rst (hunk 6 above)
Comment (neon60) [reply docs/about/compatibility/precission-support.rst]: Fixed.

File: docs/about/compatibility/precission-support.rst (hunk 7 above)
Comment (neon60) [reply docs/about/compatibility/precission-support.rst]: We do not have enough space for that.

File: docs/about/compatibility/precission-support.rst [hunk 13]
```diff
@@ -0,0 +1,544 @@
...
+      - Not supported
+      - Partial support
+      - Support without hardware acceleration
+      - Full support
+
+    * - Support icon
+      - ❌
+      - ⚠️
+      - ✔
+      - ✅
```
[the remark this answers is not in the document]
Comment (neon60) [reply docs/about/compatibility/precission-support.rst:104]: Fixed.

File: docs/about/compatibility/precission-support.rst [hunk 14]
```diff
@@ -0,0 +1,544 @@
...
+      - A 32-bit floating-point number, conforming to the IEEE 754 single precision storage format.
+    *
+      - float64
+      - ``double``
+      - A 64-bit floating-point number, conforming to the IEEE 754 double precision storage format.
+
+.. note::
+
+  * The float8 and tensorfloat32 types are internal types used in calculations in the Matrix Cores and can be stored in any type of the same size.
+  * The encodings for FP8 (E5M2) and FP8 (E4M3) natively supported by MI300 differ from the FP8 (E5M2) and FP8 (E4M3) encodings used in H100 ([FP8 FORMAT SFOR DEEP LEARNING](https://arxiv.org/abs/2209.05433)).
```
[the remark this answers is not in the document]
Comment (neon60) [reply docs/about/compatibility/precission-support.rst]: Fixed.

File: docs/about/compatibility/precission-support.md (hunk 1 above)
Comment (neon60) [reply docs/about/compatibility/precission-support.md]: Let's talk.

**Review by MKKnorr — DISMISSED** [review]
Nice job :)

**Comment by neon60** [comment]
@Rmalavally Can you review this, please? You already checked few of our data support pages. This is the main landing page, where you can access the detailed data type support pages of the projects.

**Comment by Rmalavally** [comment]
> @Rmalavally Can you review this, please? You already checked few of our data support pages. This is the main landing page, where you can access the detailed data type support pages of the projects.
> 
> We would like to extend the list of libraries, when we checked a library already.

Nicely done. Thanks for your efforts @neon60 . We could consider further enhancing the descriptions in subsequent releases.

One minor edit - I think the file name is spelt 'precission-support.md', and there are few other instances of this typo. Could you please correct it to 'precision'? Thanks so much.

**Comment by neon60** [comment]
> > @Rmalavally Can you review this, please? You already checked few of our data support pages. This is the main landing page, where you can access the detailed data type support pages of the projects.
> > We would like to extend the list of libraries, when we checked a library already.
> 
> Nicely done. Thanks for your efforts @neon60 . We could consider further enhancing the descriptions in subsequent releases.
> 
> One minor edit - I think the file name is spelt 'precission-support.md', and there are few other instances of this typo. Could you please correct it to 'precision'? Thanks so much.

Fixed.

**Review by MKKnorr — DISMISSED** [review]

File: docs/about/compatibility/data-type-support.rst [hunk 15]
```diff
@@ -0,0 +1,547 @@
...
+    :header-rows: 1
+
+    * 
+      - The support icon explanation
+      - Support icon
+    * 
+      - Not supported
+      - ❌
+    * 
+      - Partial support
```
Comment (parbenc) [inline docs/about/compatibility/data-type-support.rst:101]: this doesn't seem to be used in the document and we decided not to have this earlier. if nothing changed we should remove this

**Review by parbenc — COMMENTED** [review]
have a minor question. otherwise looks good

File: docs/about/compatibility/data-type-support.rst (hunk 15 above)
Comment (neon60) [reply docs/about/compatibility/data-type-support.rst:101]: We found one place, where it was needed.

File: docs/about/compatibility/data-type-support.rst [hunk 16]
```diff
@@ -0,0 +1,547 @@
...
+
+.. _rocm-supported-data-types:
+
+*************************************************************
+ROCm data type specifications
+*************************************************************
+
+Integral Types
+==========================================
+
```
Comment (saadrahim) [inline docs/about/compatibility/data-type-support.rst:13]: All tables should be introduced by preceding text. Similarly, all headers need to be followed by text before including a figure or table.

File: docs/data/about/compatibility/floating-point-data-types.png
Comment (saadrahim) [inline docs/data/about/compatibility/floating-point-data-types.png:1]: The two FP8s, can you differentiate between the two in naming?

**Review by saadrahim — CHANGES_REQUESTED** [review]

File: docs/about/compatibility/data-type-support.rst (hunk 16 above)
Comment (neon60) [reply docs/about/compatibility/data-type-support.rst:13]: I made the changes, asked @LisaDelaney to check the grammar.

File: docs/about/compatibility/data-type-support.rst [hunk 17]
```diff
@@ -0,0 +1,566 @@
...
+The ROCm™ components supported signed and unsigned integral types with different
+width listed with the HIP type and short description in the following table.
```
Comment (MKKnorr) [inline docs/about/compatibility/data-type-support.rst:15]:
```suggestion
The signed and unsigned integral types that are supported by ROCm™ are listed in the following table, together with their corresponding HIP type and a short description.
```

File: docs/about/compatibility/data-type-support.rst [hunk 18]
```diff
@@ -0,0 +1,566 @@
...
+      - ``int16_t``, ``uint16_t``
+      - An signed or unsigned 16-bit integer.
+    * 
+      - int32
+      - ``int32_t``, ``uint32_t``
+      - An signed or unsigned 32-bit integer.
+    * 
+      - int64
+      - ``int64_t``, ``uint64_t``
+      - An signed or unsigned 64-bit integer.
```
Comment (MKKnorr) [inline docs/about/compatibility/data-type-support.rst:41]:
```suggestion
      - A signed or unsigned 8-bit integer.
    * 
      - int16
      - ``int16_t``, ``uint16_t``
      - A signed or unsigned 16-bit integer.
    * 
      - int32
      - ``int32_t``, ``uint32_t``
      - A signed or unsigned 32-bit integer.
    * 
      - int64
      - ``int64_t``, ``uint64_t``
      - A signed or unsigned 64-bit integer.
```

File: docs/about/compatibility/data-type-support.rst [hunk 19]
```diff
@@ -0,0 +1,566 @@
...
+ 
+Floating-point Types
+==========================================
+
+The ROCm™ components supported floating-point types listed with the HIP type and
+short description in the following table:
```
Comment (MKKnorr) [inline docs/about/compatibility/data-type-support.rst:47]:
```suggestion
The floating-point types that are supported by ROCm™ are listed in the following table, together with their corresponding HIP type and a short description.
```

File: docs/about/compatibility/data-type-support.rst [hunk 20]
```diff
@@ -0,0 +1,566 @@
...
+
+  * The float8 and tensorfloat32 types are internal types used in calculations in the Matrix Cores and can be stored in any type of the same size.
+  * The encodings for FP8 (E5M2) and FP8 (E4M3) natively supported by MI300 differ from the FP8 (E5M2) and FP8 (E4M3) encodings used in H100 (`FP8 formats for Deep Learning <https://arxiv.org/abs/2209.05433>`_ ).
+  * In some AMD documents and articles float8 (E5M2) is called bfloat8.
+
+The support icon description
+==========================================
+
+The ROCm™ data support icon explanation listed in the following table. These
+icons are used at the data type support pages with the same meaning.
```
Comment (MKKnorr) [inline docs/about/compatibility/data-type-support.rst:99]:
```suggestion
The ROCm™ data support icon explanation is listed in the following table. These
icons are used at the data type support pages of the libraries with the same meaning.
```

File: docs/about/compatibility/data-type-support.rst [hunk 21]
```diff
@@ -0,0 +1,566 @@
...
+        - ✅
+        - ✅
+        - ❌
+        - ✅
+        - ✅
+
+Matrix core support
+-------------------------------------------------------------------------------
+
+The matrix core support of data types listed in the following table.
```
Comment (MKKnorr) [inline docs/about/compatibility/data-type-support.rst:213]:
```suggestion
The following table lists which data types are supported on the matrix cores of different AMD GPUs.
```

File: docs/about/compatibility/data-type-support.rst [hunk 22]
```diff
@@ -0,0 +1,566 @@
...
+        - ✅
+        - ✅
+        - ✅
+        - ✅
+        - ✅
+
+Atomic operations support
+-------------------------------------------------------------------------------
+
+The atomic operations support of data types listed in the following table.
```
Comment (MKKnorr) [inline docs/about/compatibility/data-type-support.rst:294]:
```suggestion
The support for atomic opperations for the different AMD GPUs is listed in the following table.
```

File: docs/about/compatibility/data-type-support.rst [hunk 23]
```diff
@@ -0,0 +1,566 @@
...
+  At natively not supported cases, the atomic operations can be emulated by
+  software. Such software emulated atomic operations have high negative
+  performance impact, when they frequently acces the same memory address.
+
+Data Type support in ROCm Libraries
+==========================================
+
+ROCm™ libraries support for int8, float8 (E4M3), float8 (E5M2), int16, float16,
+bfloat16, int32, tensorfloat32, float32, int64 and float64 listed in the
+following tables.
```
Comment (MKKnorr) [inline docs/about/compatibility/data-type-support.rst:383]:
```suggestion
ROCm™ libraries support for int8, float8 (E4M3), float8 (E5M2), int16, float16,
bfloat16, int32, tensorfloat32, float32, int64 and float64 is listed in the
following tables.
```

File: docs/about/compatibility/data-type-support.rst [hunk 24]
```diff
@@ -0,0 +1,566 @@
...
+
+Libraries input/output type support
+-------------------------------------------------------------------------------
+
+The ROCm™ libraries support of specific input and output data types listed in 
+the following tables. For a detailed description open the library data type
+support page.
```
Comment (MKKnorr) [inline docs/about/compatibility/data-type-support.rst:390]:
```suggestion
The ROCm™ libraries support of specific input and output data types is listed in 
the following tables. For a detailed description open the corresponding library data type
support page.
```

File: docs/about/compatibility/data-type-support.rst [hunk 25]
```diff
@@ -0,0 +1,566 @@
...
+        - ✅/✅
+        - ✅/✅
+
+
+Libraries internal calculations type support
+-------------------------------------------------------------------------------
+
+The ROCm™ libraries support of specific internal data types listed in the
+following tables. For a detailed description open the library data type support
+page.
```
Comment (MKKnorr) [inline docs/about/compatibility/data-type-support.rst:519]:
```suggestion
The ROCm™ libraries support of specific internal data types is listed in the
following tables. For a detailed description open the corresponding library data type support
page.
```

File: docs/index.md [hunk 26]
```diff
@@ -56,6 +56,7 @@ Our documentation is organized into the following categories:
 * {doc}`User/kernel space<rocm-install-on-linux:reference/user-kernel-space-compat-matrix>`
 * {doc}`Docker<rocm-install-on-linux:reference/docker-image-support-matrix>`
 * [OpenMP](./about/compatibility/openmp.md)
+* [Precission Support](./about/compatibility/data-type-support.rst)
```
Comment (MKKnorr) [inline docs/index.md:59]:
```suggestion
* [Precision Support](./about/compatibility/data-type-support.rst)
```

File: docs/sphinx/_toc.yml.in [hunk 27]
```diff
@@ -104,6 +104,8 @@ subtrees:
     title: LLVM ASan
   - file: conceptual/cmake-packages.rst
     title: Using CMake
+  - file: about/compatibility/data-type-support.rst
+    title: Precission support in ROCm libraries
```
Comment (MKKnorr) [inline docs/sphinx/_toc.yml.in:108]:
```suggestion
    title: Precision support in ROCm libraries
```

**Review by MKKnorr — CHANGES_REQUESTED** [review]

File: docs/about/compatibility/data-type-support.rst (hunk 16 above)
Comment (neon60) [reply docs/about/compatibility/data-type-support.rst:13]: @MKKnorr reviewed.

File: docs/about/compatibility/data-type-support.rst [hunk 28]
```diff
@@ -0,0 +1,564 @@
...
+        - ❌
+        - ❌
+        - ✅
+        - ✅
+
+.. note::
+
+  At natively not supported cases, the atomic operations can be emulated by
+  software. Such software emulated atomic operations have high negative
+  performance impact, when they frequently acces the same memory address.
```
Comment (MKKnorr) [inline docs/about/compatibility/data-type-support.rst:374]:
```suggestion
  performance impact, when they frequently access the same memory address.
```
Typo

**Review by saadrahim — APPROVED** [review]

**Pull request merged**